core/server/worker/src/main/java/alluxio/worker/block/management/tier/SwapRestoreTask.java

```diff
@@ -141,12 +141,15 @@ public BlockManagementTaskResult run() {
         long blockId = tierIterator.next();
         try {
           BlockMeta nextBlockFromTier = mEvictorView.getBlockMeta(blockId);
-          moveOutBytes -= nextBlockFromTier.getBlockSize();
-          if (lastTier) {
-            blocksToRemove.add(nextBlockFromTier.getBlockId());
-          } else {
-            blocksToTransfer.add(BlockTransferInfo.createMove(nextBlockFromTier.getBlockLocation(),
-                nextBlockFromTier.getBlockId(), destLocation));
+          if (nextBlockFromTier != null) {
```

Comment:
Could you log a debug message when this is null? That way we can know if the `if` block is executed. It always frustrates me when some logic should have been executed but somehow is not (or even worse, I don't know if it is or not).